lib/internal/Magento/Framework/Config/Data/ConfigDataFactory.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\Config\Data;
+
+/**
+ * Factory for ConfigData
+ * @api
+ */
+class ConfigDataFactory
```

Comment:
Sorry I missed this before but this factory should either be updated to match the standard defined in http://devdocs.magento.com/guides/v2.2/extension-dev-guide/factories.html or actually it might be possible to use the autogenerated factory here.

Reply:
Autogenerated factories are not allowed in `Magento\Framework` as far as I know. But, the best option is to commit autogenerated factory here.